### PR TITLE
parser: SQL BINDING support UNION ALL/INTERSECT/EXCEPT

### PR DESCRIPTION
**Pull request by rebelice**
### What problem does this PR solve? 
This is the part of [pull/20010](https://github.com/pingcap/tidb/pull/20010)

### Check List 

Tests 

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by lzmhhh123** [comment]
@rebelice Please resolve conflicts since `SetOprStmt` support parentheses.

File: parser.y [hunk 1]
```diff
@@ -11094,8 +11094,27 @@ CreateBindingStmt:
...
-			OriginSel:   selStmt,
-			HintedSel:   hintedSelStmt,
+			OriginNode:  selStmt,
+			HintedNode:  hintedSelStmt,
+			GlobalScope: $2.(bool),
+		}
+
+		$$ = x
+	}
+|	"CREATE" GlobalScope "BINDING" "FOR" SetOprStmt "USING" SetOprStmt
```
Comment (lzmhhh123) [inline parser.y:11104]: `SetOprStmt` don't full contain `SetOprStmt`. Could you parse SQL like:
```sql
create binding for select 1 union select 2 intersect select 3 using select 1 union select 2 intersect select 3
```
Please use `SetOprStmt1` instead it and remove `SelectStmt`.

Comment (lzmhhh123) [reply parser.y:11104]: And also please add some parser tests for binding and set operator statements.

**Review by lzmhhh123 — COMMENTED** [review]
LGTM

**Review by winoros — COMMENTED** [review]
lgtm

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**